### PR TITLE
Logger hotfix

### PR DESCRIPTION
**Pull request by dsvoronin**
* fix stackoverflowerror during InHouseInstrumentationTestRunner init

logger refactoring leftovers

* add RoundRobinService for elastic to improve stability

* enable Elastic logging for android runtime

* bump version

* use comma-separated param instead of |, because of am instrument command limitations

https://developer.android.com/studio/test/command-line

### CODE REVIEW

File: subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt [hunk 1]
```diff
@@ -61,13 +64,11 @@ class VideoCapturerImpl(
...
 
-    @SuppressLint("LogNotTimber")
     @Synchronized
-    override fun stop(): VideoCapturer.RecordResult {
-        val state = state
-        val result = when {
-            state is State.Recording -> {
-                val (videoFile, outputFile) = state
+    override fun stop(): VideoCapturer.RecordResult =
+        when (val castHelperLocalState = state) {
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt:69]: Here is no casting

File: subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/RoundRobinService.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+internal class RoundRobinService(
+    factory: (endpoint: URL) -> ElasticService,
+    endpoints: List<URL>
+) : ElasticService {
+
+    private val roundRobin =
+        RoundRobinIterable(endpoints.map { endpoint -> lazy { factory(endpoint) } })
+
+    override fun log(indexPattern: String, date: String, params: Map<String, String>): Call<ResponseBody> {
+        return roundRobin.next().value.log(
```
Comment (eugene-krivobokov) [inline subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/RoundRobinService.kt:16]: As before, I don't understand why we switch to the next destination on each invocation but not after error.

File: subprojects/common/time/src/main/kotlin/com/avito/time/DefaultTimeProvider.kt [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
+import java.util.TimeZone
 import java.util.concurrent.TimeUnit
 
+/**
+ * Used in android runtime < 26, so no java.time API yet
+ * consider using ThreeTenABP
+ */
 class DefaultTimeProvider : TimeProvider {
 
+    override val timeZone: TimeZone = TimeZone.getTimeZone("Europe/Moscow")
```
Comment (eugene-krivobokov) [inline subprojects/common/time/src/main/kotlin/com/avito/time/DefaultTimeProvider.kt:17]: Why not default timezone?
It seems that the common library shouldn't know this.

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/video/VideoCapturer.kt:69]: `Smart cast to com.avito.android.test.report.video.VideoCapturerImpl.State.Recording`

File: subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/RoundRobinService.kt (hunk 2 above)
Comment (dsvoronin) [reply subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/RoundRobinService.kt:16]: i will write a retry interceptor for that

File: subprojects/common/time/src/main/kotlin/com/avito/time/DefaultTimeProvider.kt (hunk 3 above)
Comment (dsvoronin) [reply subprojects/common/time/src/main/kotlin/com/avito/time/DefaultTimeProvider.kt:17]: yeah, you're right, let me check how a default timezone resolved first

**Pull request merged**